drivers/serial/uart_shell.c

```diff
@@ -71,14 +70,18 @@ static int cmd_uart_read(const struct shell *sh, size_t argc, char **argv)
 			shell_error(sh, "Failed to read from UART (%d)", ret);
 			return ret;
 		}
+		ret = uart_err_check(dev);
+		if (ret && ret != -ENOSYS) {
```

Comment:
`uart_err_check()` returns an `int`. A simple check for `ret != -ENOSYS` should be sufficient.